Prototypes.lua

```diff
@@ -1786,39 +1786,63 @@ WeakAuras.event_prototypes = {
     name = L["DBM Announce"],
     init = function(trigger)
       WeakAuras.RegisterDBMCallback("DBM_Announce");
-      return "";
+      local ret = "local use_cloneId = %s;"
+      return ret:format(trigger.use_cloneId and "true" or "false");
     end,
+    statesParameter = "all",
+    canHaveAuto = true,
     args = {
       {
         name = "message",
         init = "arg",
         display = L["Message"],
-        type = "longstring"
-      }
+        type = "longstring",
+        store = true
+      },
+      {
+        name = "name",
+        init = "message",
+        hidden = true,
+        test = "true",
+        store = true
+      },
+      {
+        name = "icon",
+        init = "arg",
+        store = true,
+        hidden = true,
+        test = "true"
+      },
+      {
+        name = "cloneId",
+        display = L["Clone per Event"],
+        type = "toggle",
+        test = "true",
+        init = "use_cloneId and WeakAuras.GetUniqueCloneId() or ''"
+      },
     }
   },
   ["DBM Timer"] = {
     type = "status",
     events = {
-      "DBM_TimerUpdate"
+      "DBM_TimerStart", "DBM_TimerStop", "DBM_TimerStopAll", "DBM_TimerUpdate", "DBM_TimerForce"
     },
-    force_events = "DBM_TimerUpdate",
+    force_events = "DBM_TimerForce",
     name = L["DBM Timer"],
-    init = function(trigger)
+    triggerFunction = function(trigger)
```

Comment:
Instead of Construct Function creating the function, for the DBM triggers I want to just create the trigger function myself. This patch adds that possibility by setting triggerFunction in the prototype.